system/BaseModel.php

```diff
@@ -1916,4 +1888,24 @@ protected function convertToReturnType(array $row, string $returnType): array|ob
 
         return $this->converter->reconstruct($returnType, $row);
     }
+
+    /**
+     * Checking that the ID has a rational value.
+     *
+     * Standard type checks in PHP do not fully verify the validity of ID as a primary key.
+     * Especially the values 0, '0', '' are rarely used or are invalid.
+     * Example, if you add an entry with ID=0, it will be an error for `$this->insertID`.
+     *
+     * @param array<int|string, int|string>|float|int|string|null $id
+     *
+     * @throws InvalidArgumentException
+     */
+    protected function hasAllowedValueId($id): bool
+    {
+        if (is_bool($id)) {
+            throw new InvalidArgumentException('The ID value should not be boolean.');
+        }
+
+        return ! in_array($id, [0, '0', '', 0.0], true) && (is_numeric($id) || is_string($id));
```

Comment:
Depending on what the `@param` should actually be in above, I would propose to do the check using if guard style. More verbose but clearer I believe.
```php
$pkCandidates = is_array($id) ? $id : [$id];

foreach ($pkCandidates as $candidate) {
    if (is_int($candidate) && $candidate === 0) {
        return false;
    }

    if (is_string($candidate) && in_array($candidate, ['0', ''], true)) {
        return false;
    }
}

return true;

```